src/components/Company/Offers/Manage/CompanyOffersVisibilityActions.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect, useCallback } from "react";
+import PropTypes from "prop-types";
+import {
+    IconButton,
+    Tooltip,
+} from "@material-ui/core";
+import {
+    VisibilityOff as VisibilityOffIcon,
+    Visibility as VisibilityIcon,
+} from "@material-ui/icons";
+import {
+    hideOffer as hideOfferService,
+    enableOffer as enableOfferService,
+} from "../../../../services/offerService";
+import { getHumanError } from "../../../../utils/offer/OfferUtils";
+import { useDispatch } from "react-redux";
+import { addSnackbar as addSnackbarAction } from "../../../../actions/notificationActions";
+import Offer from "../../../HomePage/SearchResultsArea/Offer/Offer";
+
+const CompanyOffersVisibilityActions = ({ offer }) => {
+
+    const dispatch = useDispatch();
+    const addSnackbar = useCallback((notification) => dispatch(addSnackbarAction(notification)), [dispatch]);
+
+    const [offerVisibilityState, setOfferVisibilityState] = useState({
+        isVisible: undefined,
+        isDisabled: undefined,
+        isBlocked: undefined,
+    });
+
+    const isHiddenOffer = offer?.isHidden;
+    const offerHiddenReason = offer?.hiddenReason;
+
+    useEffect(() => {
+        setOfferVisibilityState({
+            isDisabled: isHiddenOffer && offerHiddenReason === "ADMIN_REQUEST",
```

Comment:
I believe these values ("ADMIN_REQUEST" and "COMPANY_BLOCKED") could be stored as constants to make it easier to use them in other files as well. Maybe in the offerService?

Reply:
True